File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Data;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;

namespace summary
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = CarList.GetList();
        }

        int counter;
        int ivalue;
        double dvalue;

        private void gridView1_CustomSummaryCalculate(object sender, DevExpress.Data.CustomSummaryEventArgs e)
        {
            if (!e.IsGroupSummary) return;
            if (e.SummaryProcess == CustomSummaryProcess.Start)
            {
                ivalue = 0;
                dvalue = 0;
                counter = 0;
            }
            if (e.SummaryProcess == CustomSummaryProcess.Calculate)
            {
                if (((GridGroupSummaryItem)e.Item).FieldName == "HP")
                    if (e.GroupRowHandle % 2 == 0)
                        ivalue += (Int16)e.FieldValue;
                    else
                        ivalue += 1;
                if (((GridGroupSummaryItem)e.Item).FieldName == "Liter")
                    if (e.GroupRowHandle % 2 == 0)
                        dvalue = Math.Max(dvalue, (double)e.FieldValue);
                    else
                    {
                        dvalue += (double)e.FieldValue;
                        counter++;
                    }
            }
            if (e.SummaryProcess == CustomSummaryProcess.Finalize)
            {
                if (((GridGroupSummaryItem)e.Item).FieldName == "HP")
                    if (e.GroupRowHandle % 2 == 0)
                        e.TotalValue = ivalue;
                    else
                        e.TotalValue = ivalue;
                if (((GridGroupSummaryItem)e.Item).FieldName == "Liter")
                    if (e.GroupRowHandle % 2 == 0)
                        e.TotalValue = dvalue;
                    else
                        e.TotalValue = dvalue / counter;
            }
        }

        private void gridView1_CustomDrawRowFooterCell(object sender, DevExpress.XtraGrid.Views.Grid.FooterCellCustomDrawEventArgs e)
        {
            GridView view = (GridView)sender;
            if (e.Column.FieldName == "HP")
                if (e.RowHandle % 2 == 0)
                    e.Info.DisplayText = string.Format("Total: {0}", view.GetGroupSummaryValue(e.RowHandle, (GridGroupSummaryItem)view.GroupSummary[0]));
                else
                    e.Info.DisplayText = string.Format("Count: {0}", view.GetGroupSummaryValue(e.RowHandle, (GridGroupSummaryItem)view.GroupSummary[0]));
            if (e.Column.FieldName == "Liter")
                if (e.RowHandle % 2 == 0)
                    e.Info.DisplayText = string.Format("Max: {0}", view.GetGroupSummaryValue(e.RowHandle, (GridGroupSummaryItem)view.GroupSummary[1]));
                else
                    e.Info.DisplayText = string.Format("Avr: {0}", view.GetGroupSummaryValue(e.RowHandle, (GridGroupSummaryItem)view.GroupSummary[1]));
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Data
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid

Namespace summary
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			gridControl1.DataSource = CarList.GetList()
		End Sub

		Private counter As Integer
		Private ivalue As Integer
		Private dvalue As Double

		Private Sub gridView1_CustomSummaryCalculate(ByVal sender As Object, ByVal e As DevExpress.Data.CustomSummaryEventArgs) Handles gridView1.CustomSummaryCalculate
			If (Not e.IsGroupSummary) Then
				Return
			End If
			If e.SummaryProcess = CustomSummaryProcess.Start Then
				ivalue = 0
				dvalue = 0
				counter = 0
			End If
			If e.SummaryProcess = CustomSummaryProcess.Calculate Then
				If (CType(e.Item, GridGroupSummaryItem)).FieldName = "HP" Then
					If e.GroupRowHandle Mod 2 = 0 Then
						ivalue += CShort(Fix(e.FieldValue))
					Else
						ivalue += 1
					End If
				End If
				If (CType(e.Item, GridGroupSummaryItem)).FieldName = "Liter" Then
					If e.GroupRowHandle Mod 2 = 0 Then
						dvalue = Math.Max(dvalue, CDbl(e.FieldValue))
					Else
						dvalue += CDbl(e.FieldValue)
						counter += 1
					End If
				End If
			End If
			If e.SummaryProcess = CustomSummaryProcess.Finalize Then
				If (CType(e.Item, GridGroupSummaryItem)).FieldName = "HP" Then
					If e.GroupRowHandle Mod 2 = 0 Then
						e.TotalValue = ivalue
					Else
						e.TotalValue = ivalue
					End If
				End If
				If (CType(e.Item, GridGroupSummaryItem)).FieldName = "Liter" Then
					If e.GroupRowHandle Mod 2 = 0 Then
						e.TotalValue = dvalue
					Else
						e.TotalValue = dvalue / counter
					End If
				End If
			End If
		End Sub

		Private Sub gridView1_CustomDrawRowFooterCell(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.FooterCellCustomDrawEventArgs) Handles gridView1.CustomDrawRowFooterCell
			Dim view As GridView = CType(sender, GridView)
			If e.Column.FieldName = "HP" Then
				If e.RowHandle Mod 2 = 0 Then
					e.Info.DisplayText = String.Format("Total: {0}", view.GetGroupSummaryValue(e.RowHandle, CType(view.GroupSummary(0), GridGroupSummaryItem)))
				Else
					e.Info.DisplayText = String.Format("Count: {0}", view.GetGroupSummaryValue(e.RowHandle, CType(view.GroupSummary(0), GridGroupSummaryItem)))
				End If
			End If
			If e.Column.FieldName = "Liter" Then
				If e.RowHandle Mod 2 = 0 Then
					e.Info.DisplayText = String.Format("Max: {0}", view.GetGroupSummaryValue(e.RowHandle, CType(view.GroupSummary(1), GridGroupSummaryItem)))
				Else
					e.Info.DisplayText = String.Format("Avr: {0}", view.GetGroupSummaryValue(e.RowHandle, CType(view.GroupSummary(1), GridGroupSummaryItem)))
				End If
			End If
		End Sub
	End Class
End Namespace